sql-reportingservices-two-machines/azuredeploy.json

```diff
@@ -0,0 +1,553 @@
+      "metadata": {
+        "description": "Username for SA of the database"
+      }
+    },
+    "dbSAUserPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for SA of the database"
+      }
+    },
+    "dnsNameForPublicIP": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Globally unique DNS Name for the Public IP used to access the Virtual Machine."
+      }
+    },
+    "sqlImageVersion": {
+      "type": "string",
+      "defaultValue": "SQL2014SP1-WS2012R2",
+      "metadata": {
+        "description": "The SQL Iamge for the VM. This will pick a fully patched image of this given SQL Image"
+      }
+    },
+    "sqlImageSKU": {
+      "type": "string",
+      "defaultValue": "Enterprise",
+      "allowedValues": [
+        "Enterprise",
+        "Evaluation"
+      ],
+      "metadata": {
+        "description": "The SQL SKU, for SQL CTP Versions the only available is evaluation"  
+      }
+    },
+    "vmPrefix": {
+      "type": "string",
+      "minLength": 1,
+      "defaultValue": "SSRS",
+      "metadata": {
+        "description": "Prefix of the Virtual Machines (Prefix-RS, Prefix-Catalog)"
+      }
+    },
+    "vmSize": {
+      "type": "string",
+      "defaultValue": "Standard_D3",
+      "metadata": {
+        "description": "The size of the VM Created"
+      }
+    },
+    "vhdStorageType": {
+      "type": "string",
+      "allowedValues": [
+        "Standard_LRS",
+        "Premium_LRS"
+      ],
+      "defaultValue": "Standard_LRS",
+      "metadata": {
+        "description": "VHD storage type"
+      }
+    }
+  },
+  "variables": {
+    "sqlImagePublisher": "MicrosoftSQLServer",
+    "sqlImageOffer": "[parameters('sqlImageVersion')]",
+    "sqlImageSku": "[parameters('sqlImageSKU')]",
+    "OSDiskName": "osdiskforwindowssimple",
+    "dataDiskName": "dataDisk",
+    "nicName": "SSRSNic",
+    "addressPrefix": "10.0.0.0/16",
+    "subnetName": "Subnet",
+    "subnetPrefix": "10.0.0.0/24",
+    "vhdStorageType": "[parameters('vhdStorageType')]",
+    "publicIPAddressName": "myPublicIP",
+    "publicIPAddressType": "Dynamic",
+    "vhdStorageContainerName": "vhds",
+    "vmNameRs": "[concat(parameters('vmPrefix'), '-RS')]",
+    "vmSizeRs": "[parameters('vmSize')]",
+    "vmImageOfferRs": "[variables('sqlImageOffer')]",
+    "vmImageSkuRs": "[variables('sqlImageSku')]",
+    "vmSizeDataSource": "[parameters('vmSize')]",
+    "vmNameDataSource": "[concat(parameters('vmPrefix'), '-DataSource')]",
+    "vmSizeCatalog": "[parameters('vmSize')]",
+    "vmNameCatalog": "[concat(parameters('vmPrefix'), '-Catalog')]",
+    "virtualNetworkName": "SSRSVNET",
+    "vnetId": "[resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworkName'))]",
+    "subnetRef": "[concat(variables('vnetId'), '/subnets/', variables('subnetName'))]",
+    "vhdStorageNameRs": "[concat('vhdrs', uniqueString(resourceGroup().id))]",
```

Comment:
Do the uniquestring first in concats. It ensures the storage accounts generated from this template will be on different storage stamps